File: apps/cpp/face_detector.cc
#include <iomanip>

#include "apps_config.h"
#include "oneml/public/oneml.h"

int main(int argc, char** argv) {
#ifdef _WIN32
  _putenv_s("ONEML_CPP_MIN_LOG_LEVEL", "ERROR");
#else
  setenv("ONEML_CPP_MIN_LOG_LEVEL", "ERROR", 1);
#endif

  oneML::LicenseManager license_manager;
  license_manager.activate_trial();

  oneML::Utils utils(license_manager);
  oneML::face::FaceDetector detector(license_manager);

  // image - 1 face
  oneML::Image img;
  const std::string path(ASSETS_DIR_PATH "/face-detect-set/face/0.jpg");
  utils.read_image_cv(path, img);

  // RUN
  oneML::face::FaceDetectorResult output;
  detector.detect(img, output);

  std::cout << "Faces: " << output.get_size() << std::endl;

  for (int z = 0; z < output.get_size(); z++) {
    std::cout << "Face " << z << std::endl;

    std::vector<float> scores;
    std::vector<oneML::face::Pose> poses;
    std::vector<oneML::face::BBox> bboxes;
    std::vector<oneML::face::FaceLandmark<N_LANDMARKS>> landmarks;

    output.get_scores(scores);
    output.get_poses(poses);
    output.get_bboxes(bboxes);
    output.get_landmarks(landmarks);

    std::cout << "Score: " << std::fixed << std::setprecision(6) << scores[z] << std::endl;
    std::cout << "Pose: " << poses[z] << std::endl;
    std::cout << "BBox: [(" << bboxes[z].top << ", " << bboxes[z].left << "), (" << bboxes[z].bottom
              << ", " << bboxes[z].right << ")]" << std::endl;
    for (int l = 0; l < N_LANDMARKS; l++) {
      std::cout << "Landmark " << l << ": (" << landmarks[z].x.at(l) << ", " << landmarks[z].y.at(l)
                << ")" << std::endl;
    }
  }
}

File: apps/python/face_detector.py
import pathlib
import os

from oneML import faceAPI as api

os.environ['ONEML_CPP_MIN_LOG_LEVEL'] = "ERROR"
assets_path = os.path.join(
    str(pathlib.Path(__file__).resolve().parent.parent.parent.absolute()),
    "assets/images",
)


# helper function
def which_pose(pose):
    if pose == api.Pose_Undefined:
        return "Undefined"
    elif pose == api.Pose_Front:
        return "Front"
    elif pose == api.Pose_Left:
        return "Left"
    elif pose == api.Pose_Right:
        return "Right"
    elif pose == api.Pose_Up:
        return "Up"
    elif pose == api.Pose_Down:
        return "Down"


def main():
    manager = api.LicenseManager()
    manager.activate_trial()
    detector = api.FaceDetector(manager)
    utils = api.Utils(manager)

    img = utils.read_image_cv(
        os.path.join(assets_path, "face-detect-set/face/0.jpg")
    )

    result = detector.detect(img)

    print("Faces: " + str(result.get_size()))
    for i in range(result.get_size()):
        bbox = result.get_bboxes()[i]
        landmarks = result.get_landmarks()[i]

        print("Face " + str(i))
        print("Score: " + "".join("{:.6f}".format(result.get_scores()[i])))
        print("Pose: " + which_pose(result.get_poses()[i]))
        print(
            "BBox: [("
            + "".join("{:.6f}, {:.6f}), (".format(bbox.top, bbox.left))
            + "".join("{:.6f}, {:.6f})]".format(bbox.bottom, bbox.right))
        )

        for j in range(5):
            print(
                "Landmark "
                + "".join(
                    "{:d}: ({:.6f}, {:.6f})".format(j, landmarks.x[j], landmarks.y[j])
                )
            )

    report = detector.get_usage()
    report.to_log()


if __name__ == "__main__":
    main()